### PR TITLE
Setup, packaging, fixes for notebook 4.2

### PR DESCRIPTION
**Pull request by parente**
For #11.

* Rely on notebook 4.2+ nbextension and serverextension CLIs
* Remove extension logic from setup.py and handlers.py (atone
  for past sins!)
* Add versioneer for auto version numbering
* Add an environment.yml as a dev environment starter
* Add a Makefile to reduce typing for dev steps
* Update README install and configure steps
* Add a dev section to the README
* Fix submit URL join and encoding
* Fix submit copy path to include notebook base path explicitly
* Fetch example notebooks into notebook path, not home dir, because the two might not be the same
* Fix some UI nits in the fetch dialog

### CODE REVIEW

**Comment by parente** [comment]
Need to test a few more cases with subfolders and spaces in path / notebook names before calling it completely working.

File: nbexamples/handlers.py [hunk 1]
```diff
@@ -46,7 +44,7 @@ def list_examples(self):
         return all_examples
 
     def fetch_example(self, example_id, dest):
-        abs_dest = os.path.join(os.path.expanduser('~'), dest)
+        abs_dest = os.path.join(self.parent.notebook_dir, dest)
```
Comment (danielballan) [inline nbexamples/handlers.py]: Good catch.

File: nbexamples/static/examples.js [hunk 2]
```diff
@@ -256,7 +256,7 @@ define([
             .addClass("btn-default")
             .attr("type", "button")
             .attr("data-dismiss", "modal")
-            .text("Done")
+            .text("Cancel")
```
Comment (danielballan) [inline nbexamples/static/examples.js]: Sure. Another confusing UI point (which let me toward labeling this button "Done" in the first place) is that the model stays up after the copy notebook has been made. We should probably add a 'data-dismiss' attribute to the submit button as well.

Comment (parente) [inline nbexamples/static/examples.js]: I tried adding the data-dismiss but then the form submit doesn't happen. Doh!

I do agree we want to fix it though. Just the obvious thing didn't work.

**Comment by danielballan** [comment]
v0.1.0 has been tagged. We can settle the data-dismiss question in a future separate PR. Let me know when are you satisfied that subfolders/spaces have been tested well enough and I'll give it one last look.

**Comment by danielballan** [comment]
@parente Should I take that thumbs up and title change to mean, "Go for it"?

**Comment by parente** [comment]
Sorry. Thumbs up was for tagging. I'm still fixing some final issues with the URL encoding. Should have it in tonight and I'll give the go ahead.

File: nbexamples/static/examples.js (hunk 2 above)
Comment (parente) [inline nbexamples/static/examples.js]: I've fixed it via JS.

**Comment by parente** [comment]
OK. I think this is now GTG. I'll tackle additional issues in other PRs.

**Comment by danielballan** [comment]
Before I tag I'll try this out on our treat deployment.

**Comment by parente** [comment]
Sounds good. Thanks for testing it. Let me know if I broke anything.

**Comment by parente** [comment]
@danielballan did you get a chance to test the changes in your 4.2 environment? It'd be helpful to get a 0.2 tag so we can start on the packaging. Even if it's busted, we can iterate on fixing in 0.2.1, 0.2.2, ...

**Comment by parente** [comment]
I went ahead and tagged a v0.2.0 release for the time being so we can start on packaging. We'll tackle bugs as they arise.

**Comment by danielballan** [comment]
Fine with me. Our JupyterHub server had network hardware issues -- just figured this out today -- so I have been fighting fires instead of testing out the new packing.

**Pull request merged**